lib/controller/controller.go

```diff
@@ -334,39 +333,42 @@ func RetryPeriod(retryPeriod time.Duration) func(*ProvisionController) error {
 }
 
 // ClaimsInformer sets the informer to use for accessing PersistentVolumeClaims.
-// Defaults to using a private (non-shared) informer.
-func ClaimsInformer(informer cache.SharedInformer) func(*ProvisionController) error {
+// Defaults to using a internal informer.
+func ClaimsInformer(informer cache.SharedIndexInformer) func(*ProvisionController) error {
 	return func(c *ProvisionController) error {
 		if c.HasRun() {
 			return errRuntime
 		}
 		c.claimInformer = informer
+		c.customClaimInformer = true
 		return nil
 	}
 }
 
 // VolumesInformer sets the informer to use for accessing PersistentVolumes.
-// Defaults to using a private (non-shared) informer.
-func VolumesInformer(informer cache.SharedInformer) func(*ProvisionController) error {
+// Defaults to using a internal informer.
+func VolumesInformer(informer cache.SharedIndexInformer) func(*ProvisionController) error {
 	return func(c *ProvisionController) error {
 		if c.HasRun() {
 			return errRuntime
 		}
 		c.volumeInformer = informer
+		c.customVolumeInformer = true
 		return nil
 	}
 }
 
 // ClassesInformer sets the informer to use for accessing StorageClasses.
 // The informer must use the versioned resource appropriate for the Kubernetes cluster version
 // (that is, v1.StorageClass for >= 1.6, and v1beta1.StorageClass for < 1.6).
-// Defaults to using a private (non-shared) informer.
-func ClassesInformer(informer cache.SharedInformer) func(*ProvisionController) error {
+// Defaults to using a internal informer.
+func ClassesInformer(informer cache.SharedIndexInformer) func(*ProvisionController) error {
```

Comment:
Can we keep it cache.SharedInformer since cache.SharedIndexInformer embeds it?

Reply:
done